library/x509_crt.c

```diff
@@ -657,8 +659,11 @@ static int x509_get_subject_alt_name( unsigned char **p,
                     MBEDTLS_ERR_ASN1_UNEXPECTED_TAG );
         }
 
-        /* Skip everything but DNS name */
-        if( tag != ( MBEDTLS_ASN1_CONTEXT_SPECIFIC | 2 ) )
+        tag_mask = tag & ( MBEDTLS_ASN1_TAG_CLASS_MASK | MBEDTLS_ASN1_TAG_VALUE_MASK );
```

Comment:
What is the motivation for this masking? We're ignoring one bit of the tag here that we haven't ignored before.

Reply:
Because I asw that for `otherName` the value is `A0` == `MBEDTLS_ASN1_CONSTRUCTED |  MBEDTLS_ASN1_CONTEXT_SPECIFIC  | 0` but for DNSName is it `82` == `MBEDTLS_ASN1_CONTEXT_SPECIFIC | 2` 
I think that considering that we are interested in the type, and considering [Andres' comment](https://github.com/ARMmbed/mbedtls/pull/2530#discussion_r278226486), we can ignore even further, and get only the last byte to learnthe SAN type.

Reply:
I think that's against our current use of the `tag` field, which always contains the entire ASN.1 tag, and I don't yet see a striking reason why we should deviate from it.

Reply:
ok, I'll keep the tag in the asn1_buf , but we still need to get the type value only, which will be clearer once I rename the variable ( already done, not pushed yet)

Reply:
Just out of curiosity, what is the bit that we were not checking here? Is it the Private/Constructed bit? If so, then I definitely think we should check it.

Reply:
Yes, the bit that wasn't checked was `MBEDTLS_ASN1_CONSTRUCTED `
However, as I mentioned in https://github.com/ARMmbed/mbedtls/pull/2530#discussion_r279183010, the `otherName` contains this bit, but the `dnsName` doesn't, so we can't rely that this bit exists in the tag.
I think it's reasonbale, because `dnsName` is not a constructed type